crypto/x509v3/v3_utl.c

```diff
@@ -55,6 +56,8 @@ int X509V3_add_value(const char *name, const char *value,
     return 1;
  err:
     X509V3err(X509V3_F_X509V3_ADD_VALUE, ERR_R_MALLOC_FAILURE);
+    if (sk_allocated)
+        sk_CONF_VALUE_free(*extlist);
```

Comment:
Shouldn't this also set `*extlist` back to `NULL`? Otherwise the caller (see `X509V3_parse_list`) will double-free `*extlist` if `sk_CONF_VALUE_push` fails. Whereas the old code kept the ownership invariant working on errorsafety), even though it didn't quite roll back all changes.

PS: `X509V3_parse_list` doesn't seem to actually check the return value.

Reply:
It should, thanks!